Sway 1.6 removed support for gtk-primary-selection, but some
programs (e.g. Firefox ESR 78) still rely on it.
Since wlroots will drop it in version 0.14, we can restore
this feature on sway.
https://github.com/swaywm/sway/pull/5788
https://github.com/swaywm/wlroots/issues/2421
https://github.com/swaywm/wlroots/pull/2460
--- sway/server.c
+++ sway/server.c
@@ -15,6 +15,7 @@
 #include <wlr/types/wlr_data_control_v1.h>
 #include <wlr/types/wlr_export_dmabuf_v1.h>
 #include <wlr/types/wlr_gamma_control_v1.h>
+#include <wlr/types/wlr_gtk_primary_selection.h>
 #include <wlr/types/wlr_idle.h>
 #include <wlr/types/wlr_layer_shell_v1.h>
 #include <wlr/types/wlr_pointer_constraints_v1.h>
@@ -75,6 +76,7 @@ bool server_init(struct sway_server *server) {
 		wlr_data_device_manager_create(server->wl_display);
 
 	wlr_gamma_control_manager_v1_create(server->wl_display);
+	wlr_gtk_primary_selection_device_manager_create(server->wl_display);
 
 	server->new_output.notify = handle_new_output;
 	wl_signal_add(&server->backend->events.new_output, &server->new_output);
